bgpd/bgp_routemap.c

```diff
@@ -1189,27 +1189,38 @@ route_match_alias(void *rule, const struct prefix *prefix, void *object)
 	struct bgp_path_info *path = object;
 	char **communities;
 	int num;
+	bool found;
 
 	if (path->attr->community) {
+		found = false;
 		frrstr_split(path->attr->community->str, " ", &communities,
 			     &num);
 		for (int i = 0; i < num; i++) {
 			const char *com2alias =
 				bgp_community2alias(communities[i]);
 			if (strcmp(alias, com2alias) == 0)
-				return RMAP_MATCH;
+				found = true;
```

Comment:
Hm, won't `found` be overriden with `false` with the next community (not matched)?

Reply:
No, there are no `found = false` assignments anywhere in the loop.

Reply:
Yes, right :smile: :facepalm:

Reply:
But still, I can change the expression to be `if (!found && (strcmp(alias, com2alias) == 0))` to avoid additional comparisons when we already found the first match. I'll force-push once we get some feedback from Lou, as CI still fails even with these fixes.